Purpose: Defend against corrupt or invalid scripts living in
         /etc/rc[S0-6].d/
Fixes:   #493202
Status:  Acked by upstream.
---
Index: insserv/insserv.c
===================================================================
--- insserv.orig/insserv.c	2009-09-26 22:35:39.000000000 +0200
+++ insserv/insserv.c	2009-09-26 22:35:44.000000000 +0200
@@ -1696,6 +1696,11 @@
 	    }
 
 	    lsb = scan_script_defaults(dfd, d->d_name, override_path, &name, true, ignore);
+	    if (!name) {
+		warn("warning: script is corrupt or invalid: %s/%s%s\n", path, rcd, d->d_name);
+		continue;
+	    }
+
 	    if (!script_inf.provides || script_inf.provides == empty)
 		script_inf.provides = xstrdup(ptr);
 
